File: API/Helpers/UserParams.cs
namespace API.Helpers
{
    public class UserParams : PaginationParams
    {
        public string OrderBy { get; set; } = "date";
        public string Order { get; set; } = "asc";
        public string? SearchTerm  { get; set; } = "";
        public string Gender  { get; set; } = "";
        public int? DoctorSpecialityId  { get; set; } = null;
        public string? RoleName { get; set; }
    }
}

File: Client/src/app/models/Params/userParams.ts
import { PaginationParams } from "./paginationParams";

export class UserParams extends PaginationParams {
  orderBy = 'date';
  order = 'asc';
  gender = '';
  searchTerm = '';
  roleName = '';
  
  doctorSpecialityId: number | null = null;

  constructor(init?:Partial<UserParams>) {
    super();
    Object.assign(this, init);
  }

}
